lib/Sema/ImportResolution.cpp

```diff
@@ -264,24 +276,19 @@ class ImportResolver final : public DeclVisitor<ImportResolver> {
 ///
 /// Import resolution operates on a parsed but otherwise unvalidated AST.
 void swift::performImportResolution(SourceFile &SF) {
-  FrontendStatsTracer tracer(SF.getASTContext().Stats,
-                             "Import resolution");
-
-  // Make sure we skip adding the standard library imports if the
-  // source file is empty.
-  if (SF.ASTStage == SourceFile::ImportsResolved ||
-      SF.getTopLevelDecls().empty()) {
-    SF.ASTStage = SourceFile::ImportsResolved;
+  // If we've already performed import resolution, bail.
+  if (SF.ASTStage == SourceFile::ImportsResolved)
     return;
-  }
 
+  FrontendStatsTracer tracer(SF.getASTContext().Stats,
+                             "Import resolution");
   ImportResolver resolver(SF);
 
   // Resolve each import declaration.
   for (auto D : SF.getTopLevelDecls())
     resolver.visit(D);
 
-  SF.addImports(resolver.getFinishedImports());
+  SF.setImports(resolver.getFinishedImports());
```

Comment:
This seems ripe for requestification next.

Reply:
Yup, it's where I'm heading next :)